src/command/graph/introspect.rs

```diff
@@ -41,9 +43,8 @@ impl Introspect {
         Ok(introspect::run(GraphIntrospectInput { headers }, &client, should_retry)?.schema_sdl)
     }
 
-    pub fn exec_and_watch(&self, client: &Client, json: bool) -> RoverResult<RoverOutput> {
+    pub fn exec_and_watch(&self, client: &Client, output_opts: &OutputOpts) -> ! {
```

Comment:
`-> !` as a return signature means that this function will never end (leading to some nice compiler hints for unreachable code)

Reply:
Whoah wait -- are we doing this because `watch` is a process that we want to run for as long as the CLI is running (if called)? Is that why we use that return signature? 

And re: the unreachable code hints: are you talking about the compiler warning you if you write code below this function?

Reply:
yes and yes! you're spot on here